src/btree/col_modify.c

```diff
@@ -241,6 +241,13 @@ __wt_col_modify(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt,
 			WT_ERR(__wt_insert_serial(
 			    session, page, cbt->ins_head, cbt->ins_stack,
 			    &ins, ins_size, skipdepth, exclusive));
+
+		/*
+		 * We need to update the recno in txn op, as recno is retrieved
+		 * now only.
+		 */
+		if (logged)
+			__txn_op_modify_recno(session, cbt->recno);
```

Comment:
It doesn't look right in the layering. If I consider this function alone, I can't tell why is a cursor search operation trying to find the latest txn_op in the running transaction and updating its recno if it happens to be column store.